### DISCUSSION TITLE
[General] batch numbers not matching up

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [zkSync-Community-Hub/zksync-developers#629] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

A Cyfrin Updraft user saw this discrepency between explorerers. If you click on the Etherscan link in the ZKsync explorer link, the batch number is different. Why is that?

https://github.com/Cyfrin/foundry-full-course-cu/discussions/1943

**Comment by alhenaai** [comment]
It appears you are encountering an issue with mismatching batch numbers between the ZKsync explorer and Etherscan. This discrepancy can occur because ZKsync operates on a Layer 2 (L2) network, while Etherscan tracks the Ethereum mainnet, a Layer 1 (L1) network. Batch numbers in ZKsync (L2) are not the same as block numbers on Ethereum (L1).

For a deeper understanding of how batch and block numbers operate within the ZKsync network, you can refer to the ZKsync documentation on [Blocks and Batches](https://docs.zksync.io/zk-stack/concepts/blocks).

If you need further clarification or assistance, please let me know!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by bxpana** [comment]
Will check this out to see what's up.

**Reply by bxpana** [reply, marked as answer]
@PatrickAlphaC the `1.` values on Etherscan are the for the previous batch, but if you scroll down you'll see the matching batch number under `newBatchData`
